asmcomp/debug/dwarf_low/dwarf_operator.ml

```diff
@@ -0,0 +1,595 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                  Mark Shinwell, Jane Street Europe                     *)
+(*                                                                        *)
+(*   Copyright 2016--2018 Jane Street Group LLC                           *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+[@@@ocaml.warning "+a-4-30-40-41-42"]
+
+type implicit_value =
+  | Int of Targetint.t
+  | Symbol of string
+
+type t =
+  | DW_op_addr of string
+  | DW_op_regx of { reg_number : int; }
+  | DW_op_reg0
```

Comment:
You could simplify the code, by not differentiating between `DW_op_regx` and `DW_op_reg0` to `DW_op_reg31` and just adding a simple check for in `emit`, `print` and `encode` if the register number is between 0 and 31.

Reply:
Ah, so there is a design point here.  The types are supposed to reflect the specification very closely, which is why everything is there.  The next level up (`Dwarf_high`) deals with transformations such as the ones you suggest.

Apart from being a general point of principle, this is also a potentially useful property if this library is in the future enhanced to read DWARF, which it may well be.

Reply:
Was just a suggestion for easing the implementation, but this is also actually fine.